### PR TITLE
fix: resolve all known swap+send bugs

### PR DESCRIPTION
**Pull request by BZahory**
## **Description**

A number of minor bugs have been discovered during the official QA process of Swap+Send; this PR aims to resolve all of them without regression.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25100?quickstart=1)

## **Related issues**

[METABRIDGE-843](https://consensyssoftware.atlassian.net/browse/METABRIDGE-843)

Fixes:

## **Manual testing steps**

1. Test all issues with a send-swap label
2. Ensure they are no longer present
3. Ensure Swap+Send continues to work as expected

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx [hunk 1]
```diff
@@ -61,19 +62,14 @@ export default function AssetList({
...
-  }, [swapsBlockedTokens]);
-
   return (
     <Box className="tokens-main-view-modal">
       {tokenList.map((token) => {
         const tokenAddress = token.address?.toLowerCase();
         const isSelected = tokenAddress === selectedToken?.toLowerCase();
         const isDisabled = sendingAssetSymbol
           ? !isEqualCaseInsensitive(sendingAssetSymbol, token.symbol) &&
-            memoizedSwapsBlockedTokens.has(tokenAddress)
```
Comment (micaelae) [inline ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx:76]: why is the fallback needed?

Comment (BZahory) [reply ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx:76]: Good call – resolved

File: ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx [hunk 2]
```diff
@@ -61,19 +62,14 @@ export default function AssetList({
       hideLabel: true,
     });
 
-  const swapsBlockedTokens = useSelector(getSwapsBlockedTokens);
-  const memoizedSwapsBlockedTokens = useMemo(() => {
-    return new Set(swapsBlockedTokens);
-  }, [swapsBlockedTokens]);
```
Comment (micaelae) [inline ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx:67]: minor: any reasons why we don't store swapsBlockedTokens as a set?

Comment (BZahory) [reply ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx:67]: We can circle back to this in a follow-up PR after a discussion to keep refactors at a minimum here

**Review by ejwessel — DISMISSED** [review]

**Review by ejwessel — DISMISSED** [review]

File: ui/ducks/send/send.js [hunk 3]
```diff
@@ -1641,10 +1645,27 @@ const slice = createSlice({
...
+
+        const { quotes, gas } = draftTransaction;
         const bestQuote = quotes ? calculateBestQuote(quotes) : undefined;
+
+        const derivedGasPrice =
+          hexToDecimal(gas?.gasTotal || '0x0') > 0 &&
+          hexToDecimal(gas?.gasLimit || '0x0') > 0
+            ? new Numeric(gas.gasTotal, 16).divide(gas.gasLimit, 16).toString()
+            : undefined;
+
```
Comment (micaelae) [inline ui/ducks/send/send.js:1664]: Does the price need to be converted to gwei?

Seeing a bug where I have enough eth for gas, but the error is displayed
Screenshot 2024-06-07 at 1 48 08 PM
Screenshot 2024-06-07 at 1 48 16 PM

File: ui/components/ui/unit-input/unit-input.component.js [hunk 4]
```diff
@@ -86,6 +86,10 @@ export default class UnitInput extends PureComponent {
   };
 
   handleInputBlur = ({ target: { value } }) => {
+    setTimeout(() => {
+      this.unitInput.scrollLeft = this.unitInput.scrollWidth;
+    }, 0);
+
```
Comment (micaelae) [inline ui/components/ui/unit-input/unit-input.component.js]: what does this do? 

seeing some odd behavior when I enter a long number like `0.12414314125153525326236`. only the ending numbers are displayed on unfocus. also happens when I hit Max and my balance has a lot of trailing numbers

Screenshot 2024-06-07 at 3 35 49 PM

**Review by micaelae — APPROVED** [review]

**Review by sleepytanya — APPROVED** [review]

**Pull request merged**